src/utils/dates.js

```diff
@@ -11,9 +11,24 @@ export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
 
-export function getDaysBetweenDates(previousPurchase, currentDate) {
-	if (!currentDate) {
-		currentDate = Date.now();
+export function getDaysBetweenDates(previousPurchase, futurePurchase) {
+	const currentDate = new Date();
+	// console.log(previousPurchase, futurePurchase);
+	if (previousPurchase && futurePurchase) {
+		//console.log(`${previousPurchase && futurePurchase} previous and current`)
+		return Math.round(
+			(futurePurchase.toDate() - previousPurchase.toDate()) /
+				ONE_DAY_IN_MILLISECONDS,
+		);
+	}
+	if (!previousPurchase) {
+		// console.log(previousPurchase < currentDate);
+		return Math.round(
+			(previousPurchase.toDate() - currentDate) / ONE_DAY_IN_MILLISECONDS,
+		);
+	} else {
+		return Math.round(
+			(currentDate - previousPurchase.toDate()) / ONE_DAY_IN_MILLISECONDS,
+		);
```

Comment:
I would still advise you to think of this function as a calculation of the number of dates between two arbitrary dates

Reply:
In issue 10, it is described as `This function takes two JavaScript Dates and returns the number of days that have passed between them` -  there are ties to the meaning of the date/input

https://github.com/the-collab-lab/tcl-48-smart-shopping-list/issues/10